### DISCUSSION TITLE
questions about clusters center update when Insert new vectors/当插入大量新向量时候，怎么快速更新聚类簇的中心呢

### DISCUSSION THREAD

**Discussion by bendanzzc** [milvus-io/milvus#19114] [Q&A and General discussion] [answered]
for IVF, when the new vectors are inserted the old clusters center may not correct for new data. how to fast update center？

**Comment by yhmo** [comment, marked as answer]
Milvus manages data by "segments".

Let's say we have 1M vectors, insert into a milvus server. Milvus stores the 1M vectors into 10 segments, each segment containing 100k vectors. Then milvus build index for **each** segment, build index_1 for segment_1, index_2 for segment_2, etc. These indexes are independent, each index has many clusters and each cluster has a center.

Now we insert more vectors into milvus, the new inserted vectors are stored into a new segment, milvus build an independent index for this segment.

You call search() want to do some ANN search. Milvus use the search parameters to search across all the indexes, each index returns a topk result. Milvus combine these topk results into a final topk result and return to client.

**Reply by bendanzzc** [reply]
Thx, and I have an other question. If I call create_index() for the certain collection, whether all segments of the collection are reindexed or only the changed segments are reindexed. If I want to index some new vecs of a segment which does not reach the specified volume, what can I do to only index the segment which the new vec are inserted to?

**Reply by yhmo** [reply]
If you manually call the create_index() method. milvus will force to build index for all the persisted segments, no matter what size it is.
But since lot of small segments decrease search performance, milvus will try to combine small segments into a large segment. Once the small segments are combined, their index will be deleted and rebuild.

**Reply by bendanzzc** [reply]
Thx. So for milvus 1.xx, if some segments are indexed but do not reach the specified volume, the milvus will also try to combine these indexed small segments and reindex the combined segment. is it right?

**Reply by yhmo** [reply]
> Thx. So for milvus 1.xx, if some segments are indexed but do not reach the specified volume, the milvus will also try to combine these indexed small segments and reindex the combined segment. is it right?

In v1.x, if a segment is indexed, milvus won't combine it, unless you call drop_index()/create_index().
